parameters.go

```diff
@@ -1,16 +1,12 @@
 package winrm
 
-import (
-	"net/http"
-)
-
 // Parameters struct defines
 // metadata information and http transport config
 type Parameters struct {
 	Timeout            string
 	Locale             string
 	EnvelopeSize       int
-	TransportDecorator func(*http.Transport) http.RoundTripper
+	TransportDecorator func() Transporter
```

Comment:
@bogdanteleaga In the process of refactoring this code I was thinking on how to make this as transparent as possible without adding another parameter to NewClient or to NewClientWithParameters function so in my opinion this is a viable solution solution.